docs/aiken-integration.md

```diff
@@ -0,0 +1,516 @@
+Thus, we lose all benefits of LambdaBuffers' efficient encoding when working with Aiken's mechanisms to define types because LambdaBuffers is forced to take an extra step to translate to Aiken's inefficient encoding.
+As such, Aiken's opinionated way of encoding its data is at odds with LambdaBuffers.
+
+To resolve the mismatch in the encoding of data between the two, one could alternatively sidestep all of Aiken's methods for defining types and instead use Aiken's opaque types to alias `Data` and provide ones own constructors / record accesses as follows
+
+```rust
+use aiken/builtin as builtin
+
+pub opaque type MyRecord { data: Data }
+
+pub fn myRecord(a: Int) -> MyRecord {
+    MyRecord{ data : builtin.list_data([builtin.i_data(a)]) }
+}
+
+pub fn myRecord_a(value : MyRecord) -> Int {
+    builtin.un_i_data(builtin.head_list(builtin.un_list_data(value)))
+}
+
+// Example program:
+validator {
+    pub fn hello_world(_redeemer: Data, _scriptContext: Data) {
+        let theRecord = myRecord(69)
+
+        myRecord_a(theRecord) == 420
+    }
+}
+```
+
+Interested readers may inspect the compiled UPLC to verify that the data encoding of `MyRecord` agrees with LambdaBuffers' encoding.
+
+```purescript
+(program
+  1.0.0
+  [
+    (lam
+      i_0
+      [
+        (lam
+          i_1
+          [
+            (lam
+              i_2
+              (lam
+                i_3
+                (lam
+                  i_4
+                  (force
+                    [
+                      [
+                        [
+                          i_2
+                          [
+                            [
+                              (builtin equalsInteger)
+                              [
+                                (builtin unIData)
+                                [
+                                  i_0
+                                  [
+                                    [ i_1 (con data (I 69)) ]
+                                    (con (list data) [])
+                                  ]
+                                ]
+                              ]
+                            ]
+                            (con integer 420)
+                          ]
+                        ]
+                        (delay (con unit ()))
+                      ]
+                      (delay [ (error ) (force (error )) ])
+                    ]
+                  )
+                )
+              )
+            )
+            (force (builtin ifThenElse))
+          ]
+        )
+        (force (builtin mkCons))
+      ]
+    )
+    (force (builtin headList))
+  ]
+)
+```
+
+Note that this would most likely offer a poor user experience as this would essentially replace a large part of Aiken's language construct with our own generated functions for constructing, deconstructing, serialization / deserialization to `Data`, etc.
+
+### Aiken's packages only support fetching dependencies remotely
```

Comment:
Worth mentioning that there's a workaround. You can prepare build directory in a way to inject dependencies and trick aiken into thinking that they were fetched (see https://github.com/mlabs-haskell/uplc-benchmark/blob/master/nix/aiken/lib.nix#L83) but that's a very much an implementation detail